pkg/conduit/pipelines_parser_test.go

```diff
@@ -0,0 +1,151 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conduit
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/conduitio/conduit/pkg/foundation/assert"
+)
+
+func Test_Parser(t *testing.T) {
+	filename, err := filepath.Abs("./test/pipelines1.yml")
+	if err != nil {
+		t.Error(err)
+	}
+
+	want := PipelinesInfo{
+		Version: "1.0",
+		Pipelines: map[string]PipelineInfo{
+			"pipeline1": {
+				Status: "running",
+				Config: PipelineConfig{"pipeline1", "desc1"},
+				Processors: ProcessorInfo{
+					"pipeline1proc1": {
+						Name: "pipeline processor",
+						Type: "transform",
+						Config: ProcessorConfig{
+							Settings: map[string]string{
+								"additionalProp1": "string",
+								"additionalProp2": "string",
+							},
+						},
+					},
+				},
+				Connectors: ConnectorInfo{
+					"con1": {
+						Type:   "source",
+						Plugin: "builtin:s3",
+						Config: ConnectorConfig{
+							Name: "s3-source",
+							Settings: map[string]string{
+								"aws.region": "us-east-1",
+								"aws.bucket": "my-bucket",
+							},
+						},
+						Processors: ProcessorInfo{
+							"proc1": {
+								Name: "connector processor",
+								Type: "transform",
+								Config: ProcessorConfig{
+									Settings: map[string]string{
+										"additionalProp1": "string",
+										"additionalProp2": "string",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"pipeline2": {
+				Status: "stopped",
+				Config: PipelineConfig{"pipeline2", "desc2"},
+				Connectors: ConnectorInfo{
+					"con2": {
+						Type:   "destination",
+						Plugin: "builtin:file",
+						Config: ConnectorConfig{
+							Name: "file-dest",
+							Settings: map[string]string{
+								"path": "my/path",
+							},
+						},
+						Processors: ProcessorInfo{
+							"con2proc1": {
+								Name: "connector processor",
+								Type: "transform",
+								Config: ProcessorConfig{
+									Settings: map[string]string{
+										"additionalProp1": "string",
+										"additionalProp2": "string",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"pipeline3": {
+				Status: "stopped",
+				Config: PipelineConfig{"pipeline3", "empty"},
+			},
+		},
+	}
+
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		t.Error(err)
+	}
+
+	got, err := Parse(bytes.NewReader(data))
+	assert.Nil(t, err)
+	assert.Equal(t, want, got)
+}
+
+func Test_Parser_Duplicate_Pipeline_Id(t *testing.T) {
+	filename, err := filepath.Abs("./test/pipelines2.yml")
+	if err != nil {
+		t.Error(err)
+	}
+
+	data, err := ioutil.ReadFile(filename)
```

Comment:
Nitpick: this comment is not so important for a test, I'm still going to point it out because the upcoming PRs will need to do the same in production code and we should do it differently there. I'm talking about `ioutil.ReadFile` - this function will read the whole file into memory, then `Parse` will decode the file. It would be more efficient to just decode it while it's being read, without the need to put the whole file into memory. This can be done by passing the file directly to `Parse`, because it expects an `io.Reader` and `os.File` implements that interface:
```
file, err := os.Open("path/to/my/pipeline_file.yml")
if err != nil { ... } // handle error
defer file.Close()    // close file once we are done with it
parsedFile, err := Parse(file)
```

Reply:
@lovromazgon  this works yes, but I'm still gonna need to read the file because of the env vars, check https://github.com/ConduitIO/conduit/pull/497 
I changed the method back to accept `[]byte `because that's what's needed for the `ExpandEnv` method.
I could still pass the `io.Reader `but I'm gonna have to read it and convert it to `[]byte` just so I could get the env vars.
WDYT?

Reply:
Alright let's go with this now, even though it's not ideal, it's good enough!